src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -256,6 +256,10 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, ServerOpt
         /// <returns>The <see cref="ListenOptions"/>.</returns>
         public static ListenOptions UseHttps(this ListenOptions listenOptions, ServerOptionsSelectionCallback serverOptionsSelectionCallback, object state, TimeSpan handshakeTimeout)
         {
+            if (listenOptions.Protocols >= HttpProtocols.Http3)
```

Comment:
This catches `Http1AndHttp2AndHttp3` also, should it only be checking for `Http3`?

Reply:
You can do this to check for HTTP/3: https://github.com/dotnet/aspnetcore/blob/204ec26bac3199176d70914b41721e69959eaf32/src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs#L208

I think we want to catch any usage of HTTP/3. Having it work for 1.1 & 2, but fail as soon as a HTTP/3 request comes in would be confusing.

Reply:
```suggestion
            if (listenOptions.Protocols.HasFlags(HttpProtocols.Http3))
```